File: packages/cloud-adapters/database/interface.go
// Package database defines the cloud-adapter database axis interface.
// Bare-metal impl: CloudNativePG. Cloud impls: RDS / Cloud SQL / Azure DB.
package database

import "context"

type Handle struct {
	ID  string
	DSN string
}

type Adapter interface {
	ProvisionInstance(ctx context.Context, name, region string) (Handle, error)
	DescribeInstance(ctx context.Context, name string) (Handle, error)
	DeleteInstance(ctx context.Context, name string) error
}

File: packages/cloud-adapters/database/interface.ts
// Database adapter axis — per-tenant DB provisioning + connection
// handles. Bare-metal is CloudNativePG; cloud impls map to RDS /
// Cloud SQL / Azure DB. DSN-based connection (consumer passes the DSN
// to its pg client of choice — pool/connect surfaces stay in apps/api).

export interface DatabaseHandle {
  readonly id: string;
  readonly dsn: string;
}

export interface DatabaseAdapter {
  provisionInstance(name: string, region: string): Promise<DatabaseHandle>;
  describeInstance(name: string): Promise<DatabaseHandle>;
  deleteInstance(name: string): Promise<void>;
}
